code/controllers/subsystem/tts.dm

```diff
@@ -0,0 +1,227 @@
+#define TARGET_INDEX 1
+#define IDENTIFIER_INDEX 2
+#define START_TIME_INDEX 3
+#define REQUEST_INDEX 4
+#define MESSAGE_INDEX 5
+#define EXTRA_TARGETS_INDEX 6
+#define LANGUAGE_INDEX 7
+#define LOCAL_INDEX 8
+#define MESSAGE_RANGE_INDEX 9
+
+SUBSYSTEM_DEF(tts)
+	name = "Text To Speech"
+	wait = 0.05 SECONDS
+	priority = FIRE_PRIORITY_TTS
+	runlevels = RUNLEVEL_LOBBY | RUNLEVEL_SETUP | RUNLEVEL_GAME | RUNLEVEL_POSTGAME
+
+	/// Queued HTTP requests that have yet to be sent
+	var/list/queued_tts_messages = list()
+
+	/// HTTP requests currently in progress but not being processed yet
+	var/list/in_process_tts_messages = list()
+
+	/// HTTP requests that are being processed to see if they've been finished
+	var/list/current_processing_tts_messages = list()
+
+	/// A list of available speakers
+	var/list/available_speakers = list()
+
+	var/list/cached_voices = list()
```

Comment:
what is this list? you call it cached but we keep removing from it, seemingly anytime an audio plays?

Reply:
basically any tts request that has an active request ongoing is stored in this assoc list. The key is the identifier, which is a sha1 of the voice + filter + input.
When someone wants to request tts for a message, it checks this list first to see if there's already an ongoing request for the same voice + filter + input and adds them to the `EXTRA_TARGETS_INDEX` of the active request so that it can be played at their location when it's done processing.

Reply:
added some more detailed comments to it